extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java

```diff
@@ -118,17 +120,18 @@ public void accept(Throwable throwable) {
                 //disabled, this should be handled elsewhere and if we get to this point bad things have happened,
                 //so it is better to send a response than to hang
 
-                if (event.failure() instanceof AuthenticationCompletionException
-                        && event.failure().getMessage() != null
-                        && LaunchMode.current() == LaunchMode.DEVELOPMENT) {
-                    event.response().end(event.failure().getMessage());
+                if ((exception instanceof AuthenticationCompletionException
+                        || exception instanceof AuthenticationFailedException)
```

Comment:
Same here, `AuthenticationFailedException` with `302` will get the user redirected to OIDC